.github/workflows/build_main.yml

```diff
@@ -62,170 +67,224 @@ on:
+
+      - if: ${{ inputs.env_var_name_3 != '' }}
+        name: Set env var 3
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_3 }}"
+          echo "${{ inputs.env_var_name_3 }}=${{ secrets.env_var_value_3 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_4 != '' }}
+        name: Set env var 4
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_4 }}"
+          echo "${{ inputs.env_var_name_4 }}=${{ secrets.env_var_value_4 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_5 != '' }}
+        name: Set env var 5
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_5 }}"
+          echo "${{ inputs.env_var_name_5 }}=${{ secrets.env_var_value_5 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_6 != '' }}
+        name: Set env var 6
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_6 }}"
+          echo "${{ inputs.env_var_name_6 }}=${{ secrets.env_var_value_6 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_7 != '' }}
+        name: Set env var 7
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_7 }}"
+          echo "${{ inputs.env_var_name_7 }}=${{ secrets.env_var_value_7 }}" >> $env:GITHUB_ENV
+
+      - if: ${{ inputs.env_var_name_8 != '' }}
+        name: Set env var 8
+        run: |
+          echo "::add-mask::${{ secrets.env_var_value_8 }}"
+          echo "${{ inputs.env_var_name_8 }}=${{ secrets.env_var_value_8 }}" >> $env:GITHUB_ENV
+
+      ###############################
+      # Install required tools      #
+      ###############################
+      - name: Enable global .NET tools
+        run: |
+          echo $env:HOME\.dotnet\tools >> $GITHUB_PATH
+
+      - name: Install Roslynator
+        run: |
+          dotnet tool install --global Roslynator.DotNet.Cli
+
+      - name: Download FrendsTaskAnalyzers
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          mkdir .analyzers/tmp
+          $latest = gh release list `
+              --repo 'FrendsPlatform/FrendsTasks' `
+              --json tagName `
+              --exclude-drafts `
+              --exclude-pre-releases `
+              --jq '[.[] | select(.tagName | startswith("analyzers-")) | .tagName] | first'
+          gh release download "$latest" `
+              --repo 'FrendsPlatform/FrendsTasks' `
+              --dir '.analyzers' `
+              --pattern '*.nupkg'
+          Expand-Archive -Path '.analyzers/*.nupkg' -DestinationPath '.analyzers/tmp'
+          Get-ChildItem -Path '.analyzers/tmp' -Filter '*.dll' -Recurse | Move-Item -Destination '.analyzers' -Force
+          Remove-Item -Path '.analyzers/tmp' -Recurse -Force
+
+      ###############################
+      # Run prebuild command        #
+      ###############################
+      - if: ${{ inputs.prebuild_command != '' }}
```

Comment:
Not opposing this change of order of `name` vs `if`, but can you explain why do we need this change? The same changes in setting env vars and I am curious why this came along :)